### PR TITLE
Add create_transaction method; make indentation of queries consistent

### LINKED ISSUE #11: Request: Add a transaction
It would be useful to be able to add a transaction to an account. This would basically open the door for custom syncing of external accounts programmatically.

(For me, modifying/deleting transactions are less important. My plan is to just add a "Balance Adjustment" transaction nightly to track the value of an asset.)

### PR DESCRIPTION
**Pull request by grablair**
Adds the ability to create new transactions.

Resolves #11 

Tested locally with positive (credit) and negative (debit) numbers, with and without decimals, and with a decimal more than 2 places (logic in method rounds it to 2). Also tested with and without notes.

### CODE REVIEW

File: monarchmoney/monarchmoney.py [hunk 1]
```diff
@@ -404,6 +404,59 @@ async def get_transactions(
...
+        """
+        query = gql(
+            """
+      mutation Common_CreateTransactionMutation($input: CreateTransactionMutationInput!) {
+        createTransaction(input: $input) {
+          errors {
+            ...PayloadErrorFields
+            __typename
+          }
+          __typename
```
Comment (hammem) [inline monarchmoney/monarchmoney.py:427]: Please also return `id` for the newly created transaction.

File: monarchmoney/monarchmoney.py [hunk 2]
```diff
@@ -404,6 +404,59 @@ async def get_transactions(
...
+        merchant_name: str,
+        category_id: str,
+        notes: str = ""
+    ) -> Dict[str, Any]:
+        """
+        Creates a transaction with the given parameters
+        """
+        query = gql(
+            """
+      mutation Common_CreateTransactionMutation($input: CreateTransactionMutationInput!) {
```
Comment (hammem) [inline monarchmoney/monarchmoney.py:421]: nit: fix indentation

**Review by hammem — APPROVED** [review]
Thanks, @grablair!  Just a few minor things to fix up, but don't want to stop you from getting it done.  Really appreciate you grabbing this issue for folks.

File: monarchmoney/monarchmoney.py (hunk 1 above)
Comment (grablair) [reply monarchmoney/monarchmoney.py:427]: got it!

File: monarchmoney/monarchmoney.py (hunk 2 above)
Comment (grablair) [reply monarchmoney/monarchmoney.py:421]: Happy to do so. It looks like there may be some inconsistent indentation:

```
get_subscription_details (+ all others other than get_transactions)
-------

        query = gql(
            """
      query GetSubscriptionDetails {
        subscription {
          id
          paymentSource
          referralCode
          isOnFreeTrial
          hasPremiumEntitlement
          __typename
        }
      }
    """
        )
```

```
get_transactions
-------

        query = gql(
            """
          query GetTransactionsList($offset: Int, $limit: Int, $filters: TransactionFilterInput, $orderBy: TransactionOrdering) {
            allTransactions(filters: $filters) {
              totalCount
              results(offset: $offset, limit: $limit, orderBy: $orderBy) {
                id
                ...TransactionOverviewFields
                __typename
              }
              __typename
            }
            transactionRules {
              id
              __typename
            }
          }
    
          fragment TransactionOverviewFields on Transaction {
            id
            amount
            pending
            date
            hideFromReports
            plaidName
            notes
            isRecurring
            reviewStatus
            needsReview
            attachments {
              id
              __typename
            }
            isSplitTransaction
            category {
              id
              name
              icon
              __typename
            }
            merchant {
              name
              id
              transactionsCount
              __typename
            }
            account {
              id
              displayName
              __typename
            }
            tags {
              id
              name
              color
              order
              __typename
            }
            __typename
          }
        """
        )
```

Which one is the desired indentation? I can make them consistent across the class. My guess is the indentation used in `get_transactions` is the desired indentation.

Apologies if you're referring to some other thing I'm missing, just let me know.

**Comment by grablair** [comment]
Tested with `id` result.

```
In [1]: from monarchmoney import MonarchMoney
   ...:
   ...: import asyncio
   ...:
   ...: import os
   ...:
   ...: import json
   ...: import logging
   ...:
   ...: #logging.basicConfig(level=logging.DEBUG)
   ...:
   ...: mm = MonarchMoney(session_file = ".mm/mm_session.pickle")
   ...: asyncio.run(mm.login())
   ...:
   ...: result = asyncio.run(mm.create_transaction(
   ...:         "2023-11-07",
   ...:         "160542457753312071",
   ...:         15.326,
   ...:         "My Test Merchant 2",
   ...:         "160539862437534145"
   ...:         ))
Using saved session found at .mm/mm_session.pickle

In [2]: result
Out[2]:
{'createTransaction': {'errors': None,
  'transaction': {'id': '160643997525678328'},
  '__typename': 'CreateTransactionMutation'}}

...

In [8]: list(filter(lambda txn: txn['id'] == "160643997525678328", asyncio.run(m
   ...: m.get_transactions())['allTransactions']['results']))
Out[8]:
[{'id': '160643997525678328',
  'amount': 15.33,
  'pending': False,
  'date': '2023-11-07',
  'hideFromReports': False,
  'plaidName': None,
  'notes': '',
  'isRecurring': False,
  'reviewStatus': None,
  'needsReview': False,
  'attachments': [],
  'isSplitTransaction': False,
  'category': {'id': '160539862437534145',
   'name': 'Fun Money',
   'icon': '😜',
   '__typename': 'Category'},
  'merchant': {'name': 'My Test Merchant 2',
   'id': '160615828759320958',
   'transactionsCount': 2,
   '__typename': 'Merchant'},
...
```

File: monarchmoney/monarchmoney.py (hunk 2 above)
Comment (hammem) [reply monarchmoney/monarchmoney.py]: Thanks for the call out!  I like the way you're going in the latest commit and we'll stick with that :)

**Comment by hammem** [comment]
Please also make sure to configure [black](https://github.com/psf/black) in your Python environment to handle indentation and formatting.  

@CalvinChanCan recently enabled it against all PRs and it'll prevent any code not meeting its standard from merging cleanly.

**Comment by grablair** [comment]
Ran `black`, and made indentation consistent across the class based on the convo above.

**Pull request merged**